### ISSUE TITLE
SOURCE_BROWSE filters and xsd documentation (Origin: bugzilla #623236)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#3831]
status NEW severity *enhancement* in component *build* for *---*
Reported in version *unspecified* on platform *Other*
Assigned to: Dimitri van Heesch

Original attachment names and IDs:
- _[SourceBrowser.zip](https://sourceforge.net/projects/doxygen/files/attachments/bug623236_164998_SourceBrowser.zip/download/download)_ (ID 164998)

On 2010-06-30 19:32:22 +0000, Steve wrote:

    I need to exclude .cpp file source code from appearing in the generated
    documentation, however I have xml schema xsd files that I want to appear,
    and the when I turn SOURCE_BROWSE to NO, xsd and cpp files are filtered out.
    
    Also it would be nice if doxygen could document and create images from xsd
    schema files.

On 2010-06-30 20:43:45 +0000, Dimitri van Heesch wrote:

    Sounds like an enhancement request.
    
    Can you attach an example (source + config file in a zip) that shows what
    you are trying to do (i.e. where SOURCE_BROWSE set to NO causes the files to
    be filtered out)?

On 2010-06-30 21:18:43 +0000, Steve wrote:

    Created attachment 164998
    sample of the problem

**Comment by albert-github** [comment]
I think that a solution for this problem is:
- add settings to settings file:
  ```
  EXAMPLE_PATH = .
  EXAMPLE_PATTERNS       = *.xsd
  ```
- remove from `FILE_PATTERNS` setting: `*.xsd`
- remove from `INPUT` setting: `SampleSchema.xsd`
- create a layout file (`doxygen -l`) and add here the line
  ```
  <tab type="user" visible="no" url="SampleSchema.xsd" title="XSD Files" />
  ```

This will result in the required behavior in the 1.9.5 version.
Note:
- in the 1.9.5. version we still see the "XSD Files" in the output, in the current master (1.9.6 (2b742ad6209315d6b40ed33a6b4e967bd20639b3)) this is gone as `visible="no"` is honored.
- see also the other possibilities in respect to `tab`s of type `user` and `usergroup`.


I think the problem is solved in the mean time / fixed but not released.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
